Realm/Realm.Weaver/Analytics/Analytics.cs

```diff
@@ -0,0 +1,846 @@
+                    if (IsInRealmNamespace(instruction.Operand))
+                    {
+                        featureDict[GetInstanceAsync] = 1;
+                        return true;
+                    }
+
+                    return false;
+                }
+            },
+            {
+                nameof(GetInstance),  (instruction, featureDict, references) =>
+                {
+                    if (IsInRealmNamespace(instruction.Operand))
+                    {
+                        featureDict[GetInstance] = 1;
+                        return true;
+                    }
+
+                    return false;
+                }
+            },
+            {
+                nameof(NOT_SUPPORTED_YET),  (instruction, featureDict, references) =>
+                {
+                    featureDict[NOT_SUPPORTED_YET] = 1;
+                    return true;
+                }
+            },
+            {
+                nameof(Find), (instruction, featureDict, references) =>
+                {
+                    if (IsInRealmNamespace(instruction.Operand))
+                    {
+                        featureDict[Find] = 1;
+                        return true;
+                    }
+
+                    return false;
+                }
+            },
+            {
+                nameof(WriteAsync),  (instruction, featureDict, references) =>
+                {
+                    if (IsInRealmNamespace(instruction.Operand))
+                    {
+                        featureDict[WriteAsync] = 1;
+                        return true;
+                    }
+
+                    return false;
+                }
+            },
+            {
+                nameof(ThreadSafeReference),  (instruction, featureDict, references) =>
+                {
+                    featureDict[ThreadSafeReference] = 1;
+                    return true;
+                }
+            },
+            {
+                nameof(Add),  (instruction, featureDict, references) =>
+                {
+                    featureDict[Add] = 1;
+                    return true;
+                }
+            },
+            {
+                nameof(ShouldCompactOnLaunch),  (instruction, featureDict, references) =>
+                {
+                    featureDict[ShouldCompactOnLaunch] = 1;
+                    return true;
+                }
+            },
+            {
+                nameof(MigrationCallback),  (instruction, featureDict, references) =>
+                {
+                    featureDict[MigrationCallback] = 1;
+                    return true;
+                }
+            },
+            {
+                nameof(RealmChanged),  (instruction, featureDict, references) =>
+                {
+                    featureDict[RealmChanged] = 1;
+                    return true;
+                }
+            },
+            {
+                "SubscribeForNotifications", (instruction, featureDict, references) =>
```

Comment:
I suppose it's unlikely, but it seems like this could succeed even if it's called on non-Realm classes, am I right? 
I imagine that could be difficult to check

Reply:
Good point. I'll look into what can be done, but I'm not sure how easy it is going to be.

Reply:
I looked at this and I can't resolve the generic type of the collection. Not being able to do so makes checking for type impossible.
@nirinchev  just to have your eye on this, for example given ```System.Collections.Generic.ISet`1<!!0>``` I'd like to have a `Mono.Cecil.TypeDefinition` in order to check if it's a collection of `RealmObject`s I tried
```cs
((GenericInstanceType)(TypeSpecification)methodSpecification.Parameters[0].ParameterType).GenericArguments[0].Resolve();
```
but the resolution is null. I don't really understand why as I expected all the signatures to be compiled at this point, but mono.cecil's [test](https://github.com/jbevain/cecil/blob/master/Test/Mono.Cecil.Tests/ResolveTests.cs#L212-L221) expects the same.
@nirinchev is there something I'm missing or it's just not possible? If the latter is the case, do you know why so?

Reply:
Can we check the assembly from which `SubscribeForNotifications` comes?